src/Contract/Operation/ScanLeftable.php

```diff
@@ -18,9 +19,12 @@
 interface ScanLeftable
 {
     /**
-     * @param T|null $initial
+     * @template V
      *
-     * @return Collection<TKey, T>
+     * @param callable(V, T, TKey, Iterator<TKey, T>): V $callback
+     * @param V $initial
+     *
+     * @return Collection<int|TKey, T|V>
```

Comment:
```suggestion
     * @return Collection<TKey, V>
```
shouldn't it be like this? I'm not sure why we need to mention `int` separately, also it seems like the elements in the final collection will be of the type returned by the callback (`V`), just like in `Reduction`

Reply:
Let me double-double-double-check :)

Reply:
Nope, this is correct. The `ScanLeft` will return the initial value (V) if the `iterable` is empty, so, this is correct.

Reply:
that's what I said as well, but why `V|T` and not `V`?

Reply:
That said, there is another issue. I'm being fixing it.

Reply:
Fixed.